### DISCUSSION TITLE
can/should KSMEPlayer be used as an audio only player?

### DISCUSSION THREAD

**Discussion by nightfall708** [kingslay/KSPlayer#253] [General]
I've tried this, and it seems I have an exception related to the metal view. I'll try to fix, but the question is if this is good practice. 
I thought about implementing another player myself with `AudioEnginePlayer` and `MEPlayerItem`, but `MEPlayerItem` is not public. (`AudioEnginePlayer` is)

**Comment by kingslay** [comment]
Yes. It can. 
you can set
`KSOptions.videoDisable = false`

**Comment by nightfall708** [comment]
So it turns out `KSMEPlayer` still creates a metal view as it's required to be initialized. So my idea is to make this metal view optional i.e. `videoOutput: MetalPlayView` -> `MetalPlayView?`. 
But... the other issue is that `MediaPlayerProtocol` requires a non-optional `View` as well. This is my current direction.

**Comment by kingslay** [comment]
fix by f1e079081b281ae16ed9596e3e88ff9fc06b2b9e
